tests/webSocket-spec.coffee

```diff
@@ -0,0 +1,523 @@
+#
+# Module dependencies.
+#
+
+CONFIG              = require './support/server/config'
+protocol            = require '../src/protocol'
+ReallyError         = require '../src/really-error'
+WebSocketTransport  = require '../src/transports/webSocket'
+CallbacksBuffer     = require '../src/callbacks-buffer'
+
+options =
+  reconnectionMaxTimeout: 30e3
+  heartbeatTimeout: 3e3
+  heartbeatInterval: 5e3
+  reconnect: true
+  onDisconnect: 'buffer'
+
+describe 'webSocket', ->
+
+  describe 'initialization', ->
+
+    it 'should construct URL that matches Really URL scheme when domain is passed', ->
+      ws = new WebSocketTransport('wss://a6bcc.api.really.io', 'ibj88w5aye', options)
```

Comment:
you don't have to pass `options` in tests it's optional, when someone reads the specs he'd have the feeling it's required